signer/src/storage/postgres.rs

```diff
@@ -469,10 +469,13 @@ impl PgStore {
             "#,
         )
         .bind(chain_tip)
-        .bind(min_block_height)
+        .bind(i64::try_from(min_block_height).map_err(Error::ConversionDatabaseInt)?)
         .fetch_optional(&self.0)
         .await
-        .map_err(Error::SqlxQuery)
+        .map_err(Error::SqlxQuery)?
+        .map(BitcoinBlockHeight::try_from)
+        .transpose()
+        .map_err(Error::ConversionDatabaseInt)
```

Comment:
This error variant is supposed to be for storing into the database. Maybe add a new one, or use `Error::TypeConversion`, when reading from the database since we'll get a better error message that way.

Reply:
Sooo, for extracting from db we don't want to use this error?..

Reply:
The error message talks about converting to a value when writing to the database. That's not really this case, so we should probably use something else.

Reply:
Maybe still use this type? Like this conversion is in some sense "while writing (or reading) to db". 

If the issue is that here we don't directly write but preparing to write (do post-read actions) I don't think it is really matter, logically it is same -- we got a number from db which conflicts with our type system (or we try to write some data to db which conflicts with db type system).

If the issue is that we want separate reads and writes then 1) From name of the error I don't think it is about only reads or only writes. 2) Maaaybe we can benefit from such separation but probably just nice error message is enough

Reply:
Well, the question that I'm asking is how tough would our lives be if we needed to debug some issue here. By reusing the same error here it's not clear whether the error happened when reading or writing to the database. And we'd want to know because they'd have different implications. That's why I think we want a different error variant here, just to make our lives easier. I suggested using the `Error::TypeConversion` variant but we probably don't want to use that one either. That variant is mainly used in cases where we think that the conversion will succeed but are not confident enough to use `expect()`.

Reply:
Oh, I see, this makes sense. However, we already use `Error::ConversionDatabaseInt` for db writes for example here: https://github.com/stacks-sbtc/sbtc/blob/1719efefef588167398ac098f42dfd22d465da20/signer/src/storage/postgres.rs#L2697

But I agree with your suggestion that it can be useful to have different error types for read and write. And, I'd suggest to refactor this errors a bit, using some very explicit error names: `ConverstionDatabaseIntRead` and `ConversionDatabaseIntWrite`, but do it in a separate PR.

This will give us more explicit error naming and will not bring false feeling "`ConverstionDatabaseInt` occurs only on readings".

Do you think it can be a better choice to open an issue for such refactoring and do it in separate PR?

Reply:
> However, we already use `Error::ConversionDatabaseInt` for db writes for example here:

Yeah, it's supposed to be used only when converting to signed integers before writing to the database. Typically, we don't need one for reading because sqlx will fail when converting to the desired struct and give us an error there. Which reminds me that there may be is a better way to handle this.

Reply:
So you are ok to keep it as is for now?

Reply:
With https://github.com/stacks-sbtc/sbtc/pull/1448/commits/776797daa99c421255b29467ac9e57e8322bff18 I think we're good.